xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/saverights.vm

```diff
@@ -27,11 +27,28 @@
 #set($foundd = false)
 
 #if ($!{services.csrf.isTokenValid("$!{request.getParameter('form_token')}")})
+## Convert the user or group identifier present in the request to a DocumentReference relative to the current wiki
+## if no wiki is specified.
+#set ($userOrGroupReference = $services.model.resolveDocument($fullname, 'current'))
 #foreach($obj in $doc.getObjects($clsname)) ## XWiki.XWikiGlobalRights or XWiki.XWikiRights
-  #set($pers = "$!obj.getProperty($uorg).getValue()")
-  #if($pers.matches("${regextool.quote($fullname)},?")) ## found the rule containing that person
-    #set($levels = "$obj.getProperty('levels').getValue()") ## get the levels
-    #if($obj.getProperty('allow').getValue() == 1) ## found allow rule
+  #set ($pers = "$!obj.getValue($uorg)")
+  ## Convert the user or group identifiers contained in the current access rule to an array of DocumentReferences
+  ## in order to compare these references to the reference of the user or group for which the rights may need to
+  ## be updated. User or groups can be separated by a comma or a pipe (but not both in the same string),
+  ## so we split the string once for each.
+  #set ($userOrGroupIdentifiers = $pers.split(','))
+  #if ($$userOrGroupIdentifiers.length == 1)
+    #set ($userOrGroupIdentifiers = $ids[0].split('|'))
+  #end
```

Comment:
```suggestion
  #set ($userOrGroupIdentifiers = $stringtool.split($pers, ',|'))
```

See http://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#split-java.lang.String-java.lang.String- .

Reply:
Thank you for these change requests, which I applied in the latest commit.

Reply:
(I executed a new run of tests after the latest changes with local / global users or groups, both with single and multiple targets per rule)